plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx

```diff
@@ -147,9 +153,10 @@ export const RepoUrlPicker = (props: RepoUrlPickerProps) => {
       // set the secret using the key provided in the ui:options for use
       // in the templating the manifest with ${{ secrets[secretsKey] }}
       setSecrets({ [requestUserCredentials.secretsKey]: token });
+      setHostCredentials(state.host);
     },
     500,
-    [state, uiSchema],
+    [state, uiSchema, hostCredentials],
```

Comment:
Do you mean to add a new `state.hostCredentials` property? I cannot use the existing `state.host` in the comparison as the values change independently.

Reply:
@benjdlambert, I refactored to leverage a new `state.credentialsHost` property and removed the extra `useState`.

Reply:
Ah sorry, now I understand why you can't use the original state.  I'm thinking that it would be unfortunate to add an extra field to the type for this example, so let's revert the last commit and go back to the extra `useState` call isntead.